### PR TITLE
Merge back Release 4.3.0

### PR DESCRIPTION
**Pull request by wget**
**Summary**

If all the tests with msi, codesigning and new pipeline goes ok, we have to merge this back into master

the reason for going this way is because we didn't have time to properly test everything without doing a full release, so code is in this branch instead of master, and so we have to merge back into master.

**Additional Notes**

For separate PRs, code reviews and commetns see the following PRs:
#1044 
#1034 
#1037 
#1042 
#1030 
#1029

### CODE REVIEW

**Comment by wget** [comment]
Appveyor is expected to fail since we migrated to Circle CI. If the Circle CI build and tests are passing we will be able to do the squash. :)

File: scripts/tools.ps1 [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    # C:\Program Files\nodejs\node_modules\npm\bin
+    $progFile = ${env:ProgramFiles}
+    $npmDir = Join-Path -Path "$progFile" -ChildPath "nodejs"
+    if ([System.IO.File]::Exists("$npmDir\npm.cmd")) {
+        return $npmDir
+    }
+    return $null
+}
+
+# endregion
```
Comment (cpanato) [inline scripts/tools.ps1]: new line

File: scripts/wix311.sha [hunk 2]
```diff
@@ -0,0 +1 @@
+3f619089b46df893f55e58832ce442678fb0635f *./wix311.exe
```
Comment (cpanato) [inline scripts/wix311.sha]: new line

File: scripts/msi_installer_files_set_win64.xslt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  
+  <!-- src.: https://stackoverflow.com/a/4825010/3514658 -->
+  <xsl:template match="wi:ComponentGroup[@Id='MattermostDesktopFiles']/wi:Component">
+    <xsl:copy>
+      <xsl:attribute name="Win64">yes</xsl:attribute>
+        <xsl:copy-of select="@*"/>
+      <xsl:apply-templates />
+    </xsl:copy>
+  </xsl:template>
+</xsl:stylesheet>
```
Comment (cpanato) [inline scripts/msi_installer_files_set_win64.xslt]: new line

File: scripts/msi_installer_files_replace_id.xslt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  
+  <!-- src.: https://stackoverflow.com/a/19259647/3514658 -->
+  <xsl:template match="wi:ComponentGroup[@Id='MattermostDesktopFiles']/wi:Component/wi:File[contains(@Source,'Mattermost.exe')]">
+     <xsl:copy>
+        <xsl:attribute name="Id">MattermostDesktopEXE</xsl:attribute>
+        <xsl:copy-of select="@*[name()!='Id']"/>
+        <xsl:apply-templates />
+     </xsl:copy>
+  </xsl:template>
+</xsl:stylesheet>
```
Comment (cpanato) [inline scripts/msi_installer_files_replace_id.xslt]: new line

File: scripts/cp_artifacts.sh [hunk 5]
```diff
@@ -1,14 +1,50 @@
...
-cp "${SRC}/mattermost-desktop-${VERSION}-win-ia32.zip" "${DEST}/mattermost-desktop-${VERSION}-win32.zip"
-cp "${SRC}/mattermost-desktop-${VERSION}-win-x64.zip" "${DEST}/mattermost-desktop-${VERSION}-win64.zip"
-cp "${SRC}/mattermost-desktop-setup-${VERSION}-win.exe" "${DEST}/"
-cp "${SRC}"/mattermost-desktop-*-mac.* "${DEST}/"
-cp "${SRC}"/mattermost-desktop-*-linux-* "${DEST}/"
 cp "${SRC}"/*.yml "${DEST}/"
-cp "${SRC}"/*.blockmap "${DEST}/"
+
+# exit $SOMETHING_COPIED
+exit 0
```
Comment (cpanato) [inline scripts/cp_artifacts.sh]: new line

File: scripts/Makefile.ps1 [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+        if (!($env:COM_MATTERMOST_MAKEFILE_EXECUTION_SUCCESS)) {
+            Print-Warning "Makefile interrupted by Ctrl + C or by another interruption handler."
+        }
+        Restore-ComputerState
+        exit $exitcode
+    }
+}
+
+Main
+#endregion
```
Comment (cpanato) [inline scripts/Makefile.ps1]: new line

File: resources/osx/entitlements.mac.inherit.plist [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+<plist version="1.0">
+  <dict>
+    <key>com.apple.security.cs.allow-jit</key>
+    <true/>
+    <key>com.apple.security.cs.allow-unsigned-executable-memory</key>
+    <true/>
+    <key>com.apple.security.cs.allow-dyld-environment-variables</key>
+    <true/>
+  </dict>
+</plist>
```
Comment (cpanato) [inline resources/osx/entitlements.mac.inherit.plist]: new line

File: package.json [hunk 8]
```diff
@@ -76,4 +76,4 @@
     "webpack-dev-server": "^3.1.14",
     "webpack-merge": "^4.1.4"
   }
-}
+}
```
Comment (cpanato) [inline package.json]: new line

File: electron-builder.json [hunk 9]
```diff
@@ -92,4 +96,4 @@
   "nsis": {
     "artifactName": "${name}-setup-${version}-win.${ext}"
   }
-}
+}
```
Comment (cpanato) [inline electron-builder.json]: same here as well

**Comment by wget** [comment]
@cpanato Thanks I'll fix these :)

File: electron-builder.json (hunk 9 above)
Comment (wget) [reply electron-builder.json]: done

File: package.json (hunk 8 above)
Comment (wget) [reply package.json]: done

File: resources/osx/entitlements.mac.inherit.plist (hunk 7 above)
Comment (wget) [reply resources/osx/entitlements.mac.inherit.plist]: done

File: scripts/Makefile.ps1 (hunk 6 above)
Comment (wget) [reply scripts/Makefile.ps1]: done

File: scripts/cp_artifacts.sh (hunk 5 above)
Comment (wget) [reply scripts/cp_artifacts.sh]: done

File: scripts/msi_installer_files_replace_id.xslt (hunk 4 above)
Comment (wget) [reply scripts/msi_installer_files_replace_id.xslt]: done

File: scripts/msi_installer_files_set_win64.xslt (hunk 3 above)
Comment (wget) [reply scripts/msi_installer_files_set_win64.xslt]: done

File: scripts/wix311.sha (hunk 2 above)
Comment (wget) [reply scripts/wix311.sha]: done

File: scripts/tools.ps1 (hunk 1 above)
Comment (wget) [reply scripts/tools.ps1]: done

**Pull request merged**